Sources/Widgets/Representations/ResliceCursorContextRepresentation/index.js

```diff
@@ -0,0 +1,243 @@
+
+import vtkActor from 'vtk.js/Sources/Rendering/Core/Actor';
+import vtkLineSource from 'vtk.js/Sources/Filters/Sources/LineSource';
+import vtkMapper from 'vtk.js/Sources/Rendering/Core/Mapper';
+import vtkPolyData from 'vtk.js/Sources/Common/DataModel/PolyData';
+import vtkSphereSource from 'vtk.js/Sources/Filters/Sources/SphereSource';
+import vtkWidgetRepresentation from 'vtk.js/Sources/Widgets/Representations/WidgetRepresentation';
+
+import { RenderingTypes } from 'vtk.js/Sources/Widgets/Core/WidgetManager/Constants';
+
+// ----------------------------------------------------------------------------
+// vtkResliceCursorContextRepresentation methods
+// ----------------------------------------------------------------------------
+
+function vtkResliceCursorContextRepresentation(publicAPI, model) {
+  // Set our className
+  model.classHierarchy.push('vtkResliceCursorContextRepresentation');
+
+  // --------------------------------------------------------------------------
+  // Generic rendering pipeline
+  // --------------------------------------------------------------------------
+
+  model.mapper = vtkMapper.newInstance();
+  model.actor = vtkActor.newInstance();
+  model.mapper.setInputConnection(publicAPI.getOutputPort());
+  model.actor.setMapper(model.mapper);
+  publicAPI.addActor(model.actor);
+
+  model.pipelines = {};
+  model.pipelines.center = {
+    source: vtkSphereSource.newInstance(),
+    mapper: vtkMapper.newInstance(),
+    actor: vtkActor.newInstance(),
+  };
+  model.pipelines.axes = [];
+  // Create axis 1
+  const axis1 = {};
+  axis1.line = {
+    source: vtkLineSource.newInstance(),
+    mapper: vtkMapper.newInstance(),
+    actor: vtkActor.newInstance({ pickable: true }),
+  };
+  axis1.rotation1 = {
+    source: vtkSphereSource.newInstance(),
+    mapper: vtkMapper.newInstance(),
+    actor: vtkActor.newInstance({ pickable: true }),
+  };
+  axis1.rotation2 = {
+    source: vtkSphereSource.newInstance(),
+    mapper: vtkMapper.newInstance(),
+    actor: vtkActor.newInstance({ pickable: true }),
+  };
+  // Create axis 2
+  const axis2 = {};
+  axis2.line = {
+    source: vtkLineSource.newInstance(),
+    mapper: vtkMapper.newInstance(),
+    actor: vtkActor.newInstance({ pickable: true }),
+  };
+  axis2.rotation1 = {
+    source: vtkSphereSource.newInstance(),
+    mapper: vtkMapper.newInstance(),
+    actor: vtkActor.newInstance({ pickable: true }),
+  };
+  axis2.rotation2 = {
+    source: vtkSphereSource.newInstance(),
+    mapper: vtkMapper.newInstance(),
+    actor: vtkActor.newInstance({ pickable: true }),
+  };
+
+  model.pipelines.axes.push(axis1);
+  model.pipelines.axes.push(axis2);
+
+  // Improve actors rendering
+  model.pipelines.center.actor.getProperty().setAmbient(1, 1, 1);
+  model.pipelines.center.actor.getProperty().setDiffuse(0, 0, 0);
+
+  for (let i = 0; i < model.pipelines.axes.length; i++) {
+    Object.values(model.pipelines.axes[i]).forEach((axis) => {
```

Comment:
```
for (let i = 0; i < model.pipelines.axes.length; i++) {
    Object.values(model.pipelines.axes[i]).
```
->
``` 
model.pipelines.axes.forEach((axis) => {
  Object.values(axis).forEach((lineOrRotationHandle) => {
...
}
```

Reply:
If you want speed, the first one will be faster.

Reply:
So maybe we can leave it as it is now ?

Reply:
Yes, you can leave it as is now, it was just "Bonus".

I find `forEach()` more readable and less error prone than a `for` loop.

@jourdain, performance not being an issue here, is there a best practice in VTK-JS?

Reply:
I tend also to prefer `.map()`, `.forEach()` but when things are in tight loop, the `for(let i = 0; i < array.length; i++)` are faster especially if the lambda function don't already exist.

I would enforce readability unless we need speed. More of the rendering code should look very much like C++. For the rest, it could be more like JS.

I'm on the same boat as you @finetjul. I just wanted to share that small detail since you had "perf" in one of your MR title at some point.